src/game/boe.graphics.cpp

```diff
@@ -860,6 +860,13 @@ void draw_terrain(short	mode) {
 			
 			if(is_town() || is_combat())
 				draw_items(where_draw);
+			
+			int max_dim = is_town() ? int(univ.town->max_dim) : 96;
+			if (where_draw.x<0 || where_draw.x>=max_dim || where_draw.y<0 || where_draw.y>=max_dim) {
+				place_road(q,r,where_draw,false);
```

Comment:
?

Why are you placing a road when it's out-of-bounds?

Reply:
I guess I have already protected place_road to avoid the problem when it is called when it is out of bounds. But yes, it would be better to decide what to do in this case and not call another function.